97.md

```diff
@@ -0,0 +1,92 @@
+
+# Recommendation Subscriptions list
+
+A special event with kind `10020` "Recommendation Subscription List" is defined as a [NIP-16 - Replaceable Events](16.md) of public and/or private `p` tags, one for each of the subscribed services the author intends to use. Each tag entry should contain the public key for the service provider and a relay URL where the provider respond to the recommendation requests.
+
+For example:
+
+```json
+{
+  "kind": 10020,
+  "tags": [
+    ["p", "91cf9..4e5ca", "wss://provider1.com/"],
+    ["p", "14aeb..8dad4", "wss://provider2.com/nostr"],
+    ["p", "612ae..e610f", "ws://provider3.com/ws"]
+  ],
+  "content": "<private list>",
+  ...other fields
+}
+```
+
+Providers can offer as many algorithms as they want as long as they offer one per public key. The public key's NOSTR profile SHOULD describe the service for humans. 
+
+# Recommendation Request
+
+Clients can request recommendations at will using a kind `20020` ephemeral event. `p` tags MUST contain the group of users to make the recommendation for as well as the pub key of the service provider. `.content` MUST contain a json-serialized object with a `filters` property whose value is an array of regular NOSTR `REQ` filters ([NIP-1 - Basic Protocol](1.md)) that describe the type and constraints of the requested recommendation events.
+
+```json
+{
+  "kind": 20020,
+  "tags": [
+    ["p", "<service pubkey>"],
+    ["p", "91cf9..4e5ca", "wss://relay1.com"],
+    ["p", "30223..14aeb", "wss://relay2.com"],
+  ],
+  "content": "{ \"filters\": [<FILTER1>, <FILTER2>, ...]}",
+  ...other fields
+}
+```
+
+Clients may specify several recommendation filters, i.e. `[{ "kinds": [1, 1065]}, { "#p": ["91cf9..4e5ca"], "limit": 200 }, ...]`. Clients may include `kinds`, `tags` and other filter fields to restrict recommendation results to particular event kinds. 
+
+Providers SHOULD interpret the group of users as watching the feed together. If individual users within the group have safety filters in their profiles (e.g. parents and kids together), providers SHOULD apply the sum of safety filters to the response. 
+
+Providers SHOULD interpret the recommendation request to the best of their ability and return events that they deem interesting to the requesting group. Recommendations can also include trending events as long as they are deemed of interest.
+
+Clients MAY send requests to any provider or to just the select few the user is interested in at the moment.
+
+# Recommendation Response
+
+Providers respond with an ephemeral event with kind `20021`. This event MUST contain the tags to the recommended `e` event ids, `a` addresses, `p` profiles, `r` urls and/or `t` hashtags followed by a relay address to find them. An optional normalized score from 0 (least interesting) to 1 (most interesting) SHOULD be added as the 4th element of each tag. The `p` tag of the requester and the `e` tag for the requesting event SHOULD be added as well. 
```

Comment:
Must the Recommendation Response be a response to a previous Recommendation Request? I think it's useful that  when users have subscribed to a recommendation provider successfully, the provider could push recommendation event to users continuously (a push-style recommendation).

If the push-style is not always the case, we can have a configuration field in subscription event that allows users to explicitly specify what they need.

Reply:
If we turn the request into a registration event, then I think it's better if these events become simple replaceable events (non-ephemeral). In that way, a client can register it's touch points and just download the recommendations using regular filters (as opposed to a request/reponse protocol)

The issue is that these registrations are client dependent. The request's filters is defined by the app (by the UI), not by the client. Apps can have n registrations for different parts of the app.